src/main/diagnostics/steps/step3.serverConnectivity.ts

```diff
@@ -24,7 +25,7 @@ const run = async (logger: ElectronLog): Promise<DiagnosticStepResponse> => {
                 throw new Error(`Invalid server configuration. Server Url: ${server.url}, server name: ${server.name}`);
             }
 
-            const serverOnline = await isOnline(logger, `${server.url}/api/v4/system/ping`);
+            const serverOnline = await isOnline(logger, parseURL(`${server.url}/api/v4/system/ping`)?.toString());
```

Comment:
If the `parseUrl` here returns `undefined`, will it get defaulted above in the `isOnline` function? Do we need to check for `undefined`?

Reply:
Not in this case no, it's very unlikely that this is `undefined` but if it ever is (would only happen if someone tried to break something) the default case would catch it from crashing.

Reply:
I'm not familiar what this is trying to do just want to make sure that is OK to default to `'https://community.mattermost.com/api/v4/system/ping' in the rare case it's an invalid url here

Reply:
Yeah that's fine, this is all part of a health check in case users experience issues. Worst case scenario it pings `community`, which is only an indication that their server URL is wrong/unavailable. It's actually potentially helpful for it to work this way :P

Reply:
Trying to understand the difference here, the use of `parseURL` here is only to validate if its a current URL, right? If so, are we including all the basic stuff such as removing trailing slashes, whitespaces, missing protocols, etc?

Reply:
`parseURL` is here to remove extra slashes, since we assume that one is needed before the `api` route. But often the slash is already there so we run it through `parseURL` to make sure it's all correct.

The whitespace and missing protocols should already all be taken care of at this point before the server was even added.